neural_compressor/utils/utility.py

```diff
@@ -918,6 +920,8 @@ def get_number_of_sockets() -> int:
     cmd = "lscpu | grep 'Socket(s)' | cut -d ':' -f 2"
     if sys.platform == "win32":
         cmd = 'wmic cpu get DeviceID | find /c "CPU"'
+    elif sys.platform == "darwin":
+        cmd = "sysctl -n machdep.cpu.core_count"
```

Comment:
technically i never tested this duplicate looking fn

Reply:
Yes I think it is just a duplicate method, but to keep this PR clean, we just keep it.